components/server/src/projects/projects-service.ts

```diff
@@ -95,9 +95,17 @@ export class ProjectsService {
         if (projects.length > 0) {
             throw new Error("Project for repository already exists.");
         }
+        // If the desired project slug already exists in this team or user account, add a unique suffix to avoid collisions.
```

Comment:
Maybe I am paranoid here, but wouldn't it be safer against collisions the other way around:
1) Create the project
2) Read user existing projects and increment the suffix until unique

Reply:
Are you worried about two projects with the same name being added at the exact same time?

Reply:
Kind of. As the slug yields the URI, a collision would not be nice. Maybe this is not so important for projects within teams, but more likely for teams

Reply:
For teams, team creation fails if there is a collision. The code is here:

https://github.com/gitpod-io/gitpod/blob/5d751e330d37680e6b6ab5cf6ef7b5e9496f71be/components/gitpod-db/src/typeorm/team-db-impl.ts#L125-L144